src/astUtils/visitors.ts

```diff
@@ -43,6 +47,25 @@ export function walk<T>(owner: T, key: keyof T, visitor: WalkVisitor, options: W
 
     //notify the visitor of this element
     if (element.visitMode & options.walkMode) {
+
+        if (options.walkMode & WalkMode.visitComments) {
+            const leadingTrivia = element.getLeadingTrivia();
+            for (let i = 0; i < leadingTrivia.length; i++) {
+                const trivia = leadingTrivia[i];
+                if (trivia.kind === TokenKind.Comment) {
+                    const result = (visitor as unknown as WalkTokenVisitor)?.(trivia, element as any, leadingTrivia, i);
```

Comment:
Not really happy with the typing here… but I’m not sure how to deal with this nicely. Comments are a special case.